test/int/featureBasedSuits/pauseOverlayIsShown.test.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { testUsing } from '../fixtures/testUsing';
+import { TestProjectSpec } from '../framework/frameworkTestSupport';
+import { LaunchProject } from '../fixtures/launchProject';
+import { launchArgs } from '../testSetup';
+import { expect } from 'chai';
+import * as _ from 'lodash';
+import { utils } from 'vscode-chrome-debug-core';
+
+// There is no way to validate whether we are showing the paused overlay with puppeteer, so we look into the debug-adapter
+// log and see if we sent the proper Overlay.setPausedInDebuggerMessage message
+async function latestPausedOverlay(): Promise<string | undefined> {
+    // Wait a little to give the log file time to get written...
+    // Warning: If this test starts failing because 500 ms being too little time, we should change the logic to read the file, and retry a few times to see if the assertion passes eventually
+    // If that doesn't work either, we'll need to do something less hacky like implementing a sniffer or proxy of the protocol, and get the information directly from there instead of
+    // reading it from a file
+    await utils.promiseTimeout(undefined, 500);
+
+    const logFilePath = launchArgs().logFilePath!;
+    const logFileContents = await utils.readFileP(logFilePath);
+    const lines = logFileContents.split('\n');
+    const lastEvent = _.findLast(lines, line => line.indexOf('Overlay.setPausedInDebuggerMessage') >= 0);
```

Comment:
I don't understand why you have to get it this way, isn't the test framework receiving all debug protocol traffic from the DA?

Reply:
Never mind, this is a message to chrome